qa/integration/services/service_container.rb

```diff
@@ -0,0 +1,77 @@
+require_relative "service"
+require "docker"
+
+# Represents a service running within a container.
+class ServiceContainer < Service
+
+  # Base image used by all service containers.
+  @@base_image = nil
+
+  def initialize(name, settings)
+    super(name, settings)
+
+    @base_image_context = File.expand_path("../dockerized", __FILE__)
+
+    @image_context = File.join(@base_image_context, @name)
+
+    # Options to create the container.
+    @container_create_opts = {}
```

Comment:
Removed in the last patch. I was going to used it in a ```@container.kill``` call but I decided to use ```stop```.